### PR TITLE
FORMS-5836 - Add FastTrack Analytics support to AF2 headful

### PR DESCRIPTION
**Pull request by sauraku**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/af-clientlibs/core-forms-components-runtime-all/js/datalayer.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        ERROR: "Validation errors",
+        FIELD: "Field visits",
+        ABANDON: "Abandoned forms",
+        HELP: "Help views",
+        TIMESPENT: "Time Spent",
+        SAVE: 'Drafts' // TODO: implement when portal is available for CC
+    }
+
+    const getFormName = () => {
+        const guidePath = guideBridge.getFormModel().getState().properties['fd:path'].substring(0,guideBridge.getFormModel().getState().properties['fd:path'].indexOf('/jcr:content/guideContainer'));
```
Comment (nit23uec) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/af-clientlibs/core-forms-components-runtime-all/js/datalayer.js:38]: what is the definition of form name?
 there is no such thing as name in the schema.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/af-clientlibs/core-forms-components-runtime-all/js/datalayer.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        const formPath = guidePath.substring(0,guidePath.lastIndexOf('/'));
+        const name = guidePath.substring(guidePath.lastIndexOf('/') + 1)
+        return name + '(' + formPath + ')';
+    }
+
+    const _getEventInfo = (_eventName, _fieldName, _fieldType, _panelName,_formName ) => {
+        // global variables to be used in abandon event which can be triggered anytime
+        formName = _formName;
```
Comment (nit23uec) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/af-clientlibs/core-forms-components-runtime-all/js/datalayer.js:46]: why is the field label bing passed as field name here? It is making code less readable.

File: ui.frontend/src/view/FormFieldBase.js [hunk 3]
```diff
@@ -184,6 +190,10 @@ export default class FormFieldBase extends FormField {
         return this.parentView.getModel().name;
     }
 
+    #getPanelTitle() {
+        return this.parentView.getModel().label ? this.parentView.getModel().label.value: "Root Panel";
```
Comment (nit23uec) [inline ui.frontend/src/view/FormFieldBase.js:194]: there is no such thing as Root Panel in the form.
we should not have have logic of getting panel title here. pls get it from event payload or introduce it at level of event payload inside guidebridge.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/af-clientlibs/core-forms-components-runtime-all/js/datalayer.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    const getFormName = () => {
+        const guidePath = guideBridge.getFormModel().getState().properties['fd:path'].substring(0,guideBridge.getFormModel().getState().properties['fd:path'].indexOf('/jcr:content/guideContainer'));
+        const formPath = guidePath.substring(0,guidePath.lastIndexOf('/'));
+        const name = guidePath.substring(guidePath.lastIndexOf('/') + 1)
+        return name + '(' + formPath + ')';
+    }
+
+    const _getEventInfo = (_eventName, _fieldName, _fieldType, _panelName,_formName ) => {
```
Comment (nit23uec) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/af-clientlibs/core-forms-components-runtime-all/js/datalayer.js:44]: pls us `#` to denote private functions as required by ECMA6

**Review by nit23uec — CHANGES_REQUESTED** [review]
inline comments.

**Comment by sauraku** [comment]
@nit23uec, I'll address all your concerns on this PR: https://github.com/adobe/aem-core-forms-components/pull/557
Merging https://github.com/adobe/aem-core-forms-components/pull/557 and https://github.com/adobe/aem-core-forms-components/pull/556 into single PR.
closing this PR.

**Pull request closed without merging**